### PR TITLE
Move service discovery to objectsmanager and some clean up (QC-193, QC-189)

### PR DESCRIPTION
**Pull request by Barthelemy**

### CODE REVIEW

**Comment by knopers8** [comment]
The application is not blocked anymore, but I can see "[30270]: bind: Address already in use" in the logs.

**Comment by Barthelemy** [comment]
@knopers8 Are you sure it is linked to the Service Discovery ? Do you have something running on port 7777 ? (I think that it is the default)

**Comment by knopers8** [comment]
I have checked, nothing is running on that port.

**Comment by knopers8** [comment]
I am pretty sure it is connected somehow to Service Discovery. When it was getting stuck before, it was always stuck in that place.

**Comment by Barthelemy** [comment]
Should I merge nevertheless ? the test failure is unrelated.

**Comment by awegrzyn** [comment]
I can take a look at it today.

**Comment by knopers8** [comment]
It improves the situation, so I am fine with merging this.

File: Framework/src/ObjectsManager.cxx [hunk 1]
```diff
@@ -24,9 +24,12 @@ using namespace std;
 namespace o2::quality_control::core
 {
 
-ObjectsManager::ObjectsManager(TaskConfig& taskConfig, std::shared_ptr<ServiceDiscovery> serviceDiscovery) : mTaskName(taskConfig.taskName), mServiceDiscovery(serviceDiscovery), mUpdateServiceDiscovery(false)
+ObjectsManager::ObjectsManager(TaskConfig& taskConfig) : mTaskName(taskConfig.taskName), mUpdateServiceDiscovery(false)
 {
   mMonitorObjects.SetOwner(true);
+
+  // register with the discovery service
+  mServiceDiscovery = std::make_unique<ServiceDiscovery>("http://consul-test.cern.ch:8500", mTaskName);
```
Comment (awegrzyn) [inline Framework/src/ObjectsManager.cxx]: Maybe it's better to move the URL to config file?

Comment (Barthelemy) [reply Framework/src/ObjectsManager.cxx]: yes, let me do that

Comment (Barthelemy) [reply Framework/src/ObjectsManager.cxx]: Done

**Comment by Barthelemy** [comment]
We duplicate monitoring and consul url in all json files. We should see how to do it better. For now it will do. 
@awegrzyn The test testObjectsManager::invalid_url_test shows that the ServiceDiscovery hangs if I provide a wrong url. It hangs in the destructor of ServiceDiscovery (during join) when ObjectsManager exits.

**Comment by awegrzyn** [comment]
Sure but when I just run `o2-qc-run-basic` it does not happen, what argument shall I pass in order to set URL?

**Comment by Barthelemy** [comment]
@awegrzyn I think that it would be better to work with the test because it excludes the DPL. 
I'll pass by your office.

**Comment by Barthelemy** [comment]
@awegrzyn To run the test do 
```
cd alice/sw/BUILD/QualityControl-latest/QualityControl
./bin/testObjectsManager
```
It hangs (even after your commit).

**Comment by Barthelemy** [comment]
@awegrzyn works well! Thank you
I have fixed the formatting and wait for the tests. If everything is green, I will merge it.

**Comment by awegrzyn** [comment]
Ok, but we need to verify whether it works well in the real scenario.

**Comment by Barthelemy** [comment]
Yes, but the plan is to merge , then release, then let you play with it.

**Pull request merged**